core/src/main/scala/org/apache/spark/util/UninterruptibleThread.scala

```diff
@@ -92,11 +110,27 @@ private[spark] class UninterruptibleThread(
    * interrupted until it enters into the interruptible status.
    */
   override def interrupt(): Unit = {
-    uninterruptibleLock.synchronized {
-      if (uninterruptible) {
-        shouldInterruptThread = true
+    if (uninterruptibleLock.synchronized {
```

Comment:
code style nit: shall we move the if condition into a method like `def shouldDoInterruption`?

Reply:
@cloud-fan is this covered by code style guidelines. If not, I'll make the change in case other changes are necessary, otherwise, let's keep it this way.

Reply:
Agree, refactoring into appropriate method(s) improves clarity and readability

Reply:
Neither synchronized blocks are very long. For example https://github.com/apache/spark/blob/ef336ad51ba62e64a77896b9da12791051fa92cb/core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala#L773 is much longer.

I'll make the change if another revision is necessary.

Reply:
It's unfortunate that not all the existing Spark codebase follows the best practice, but let's make sure the new code does.

Reply:
OK

Reply:
Let's not get blocked by a code style nit. @vrozov If you don't mind I can help to fix the code style via the Github UI and then we can move forward.

Reply:
Or I can merge it as it is and open a follow-up PR to fix the code style. Which way do you prefer?

Reply:
@cloud-fan Please wait. I was working on a different PR that I am now ready to open. I will fix the code style on Monday.

Reply:
@cloud-fan @mridulm Introduced `UninterruptibleLock` class with 2 helper methods.